zkevm-circuits/src/evm_circuit/execution/stop.rs

```diff
@@ -197,10 +203,16 @@ mod test {
                 },
                 |block, _tx| block.number(0xcafeu64),
             )
-            .unwrap()
-            .into()
+            .unwrap();
+
+            CircuitTestBuilder::empty()
+                .test_ctx(ctx)
+                .config(BytecodeTestConfig {
+                    enable_state_circuit_test: true,
+                    ..Default::default()
+                })
+                .run();
```

Comment:
the name `BytecodeTestConfig` seems wired now. Let me think of bytecode circuit.. Rename it in this PR directly?

Reply:
Makes sense to me to have enabled the `state` tests always. Same for `evm` probably. With the builder pattern you can always pass no checks for some of the circuits like `Box::new(|prover| //do nothing ))`

Reply:
I'd be careful with that.  It's true that having the State and EVM circuit corresponding mock provers always created would work, and would remove the need for the config; but then we would be doing the config and synthesize step for circuits that are not checked at all, and this will have a penalty in the running time of the unit tests.

Reply:
What  I can do is add a `Target` struct to the builder. And that one is the one which tells the builder which tests to run. That would make easier to adapt other circuit to this pattern for example.

Reply:
Got ridd of `BytecodeTestConfig` in 0dc7a9bb